02-express-tutorial/app.js

```diff
@@ -1 +1,58 @@
-console.log('Express Tutorial')
+const express = require('express');
+const app = express();
+const { products } = require('./data');
+
+app.use(express.static('./public'));
+
+app.get('/api/v1/test', (req, res) => {
+  res.json({ message: 'It worked!' });
+});
+
+app.get('/api/v1/products', (req, res) => {
+  res.json(products);
+});
+
+app.get('/api/v1/products/:productID', (req, res) => {
+  const idToFind = parseInt(req.params.productID);
+  const product = products.find(p => p.id === idToFind);
+
+  if (!product) {
+    return res.status(404).json({ message: 'That product was not found' });
+  }
+
+  return res.json(product);
+});
+
+app.get('/api/v1/query', (req, res) => {
+  const { search, limit, priceLimit } = req.query;
+  let foundedProducts = [...products];
+
+  if (search) {
+    foundedProducts = foundedProducts.filter(product => {
+      return product.name.startsWith(search);
+    });
+  }
+
+  if (limit) {
+    foundedProducts = foundedProducts.slice(0, parseInt(limit));
+  }
+
+  if (priceLimit) {
+    foundedProducts = foundedProducts.filter(product => {
+      return product.price <= parseInt(priceLimit);
+    });
```

Comment:
This is purely a suggestion but you could also use the implicit return of the function when you have simple return values.

`const foundedProducts = foundedProducts.filter((product) => product.price <= parseInt(priceLimit))`

This is from MDN on [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions):

```
Arrow functions can have either an expression body or the usual block body.

In an expression body, only a single expression is specified, which becomes the implicit return value. In a block body, you must use an explicit return statement.

JS
Copy to Clipboard
const func = (x) => x * x;
// expression body syntax, implied "return"

const func2 = (x, y) => {
  return x + y;
};
// with block body, explicit "return" needed
```

Reply:
I know about this type of return, but the second one is always easier for me to understand. But you are absolutely right - you need to relearn and use implicit return for simple return values.